cfn-resources/teams/cmd/resource/resource.go

```diff
@@ -59,25 +62,25 @@ func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler
 	projectID := cast.ToString(currentModel.ProjectId)
 	if teamID == "" {
 		// create new team in organization
-		teamRequest := mongodbatlas.Team{
+		teamResponse, resp, err := atlasV2.TeamsApi.CreateTeam(context.Background(), orgID, &admin.Team{
 			Name:      cast.ToString(currentModel.Name),
 			Usernames: currentModel.Usernames,
-		}
-		teamResponse, resp, err := client.Teams.Create(context.Background(), orgID, &teamRequest)
+		}).Execute()
+
 		if err != nil {
-			return progressevents.GetFailedEventByResponse(fmt.Sprintf("unable to create team %v", err), resp.Response), nil
+			return progressevents.GetFailedEventByResponse(fmt.Sprintf("unable to create team %v", err), resp), nil
 		}
-		teamID = teamResponse.ID
+		teamID = util.SafeString(teamResponse.Id)
 		currentModel = convertTeamToModel(teamResponse, currentModel)
 	}
 
 	// add existing team or newly created team to project if project id exist in the request
 	if projectID != "" && len(currentModel.RoleNames) > 0 {
-		createRequest := []*mongodbatlas.ProjectTeam{{
-			TeamID:    teamID,
+		createRequest := []admin.TeamRole{{
+			TeamId:    &teamID,
 			RoleNames: currentModel.RoleNames,
 		}}
-		_, _, err := client.Projects.AddTeamsToProject(context.Background(), projectID, createRequest)
+		_, _, err := atlasV2.TeamsApi.AddAllTeamsToProject(context.Background(), projectID, &createRequest).Execute()
 		if err != nil {
```

Comment:
[refenrece] The error-handling strategy for cloud formation is good. 
In Terraform we are printing the whole error object responses in CLI we are handling some of the specific error codes with custom messages:
https://github.com/mongodb/atlas-sdk-go/blob/main/docs/doc_2_error_handling.md#error-handling

Reply:
thanks for sharing this helper function, used it for handling a specific error code in this resource migration.